docs/migrations/v3-to-v4.md

```diff
@@ -18,14 +18,72 @@ No genesis or in-place migrations required when upgrading from v1 or v2 of ibc-g
 
 ## Chains
 
-### IS04 - Channel 
+### ICS04 - Channel 
 
 The `WriteAcknowledgement` API now takes the `exported.Acknowledgement` type instead of passing in the acknowledgement byte array directly. 
 This is an API breaking change and as such IBC application developers will have to update any calls to `WriteAcknowledgement`. 
 
 The `OnChanOpenInit` application callback has been modified.
 The return signature now includes the application version as detailed in the latest IBC [spec changes](https://github.com/cosmos/ibc/pull/629).
 
+### ICS27 - Interchain Accounts
+
+The `RegisterInterchainAccount` API has been modified to include an additional `version` argument. This change has been made in order to support ICS29 fee middelware, for relayer incentivization of ICS27 packets.
+Consumers of the `RegisterInterchainAccount` are now expected to build the appropriate JSON encoded version string themselves and pass it accordingly. 
+This should be constructed within the interchain accounts authentication module which leverages the APIs exposed via the interchain accounts `controllerKeeper`. 
+
+The following code snippet illustrates how to construct an appropriate interchain accounts Metadata and encode it as a JSON bytestring:
+
+```go
+icaMetadata := icatypes.Metadata{
```

Comment:
Since we have a constructor function (`NewMetadata`) available, shouldn't we encourage developers to use it?

Reply:
Your arguments sound reasonable to me, so I am fine leaving as it is. :)